lib/tapioca/dsl/compilers/active_record_associations.rb

```diff
@@ -121,6 +146,50 @@ def initialize(class_name)
           end
         end
 
+        class AssociationTypeOption < T::Enum
+          extend T::Sig
+
+          enums do
+            Nilable = new("nilable")
+            Persisted = new("persisted")
```

Comment:
Docs talk about an `untyped` option as well and I think it's good to add and test.

Reply:
Apologies - that was a bad copy / paste and an unintended inclusion. I've updated the docs. The two options - nilable and persisted - have tests.